crates/matrix-sdk-base/src/client.rs

```diff
@@ -824,11 +808,16 @@ impl BaseClient {
         room_id: &RoomId,
         response: &api::membership::get_member_events::v3::Response,
     ) -> Result<MembersResponse> {
-        let members: Vec<MemberEvent> = response
+        let members: Vec<_> = response
             .chunk
             .iter()
-            .filter_map(|e| e.deserialize().ok().and_then(|e| MemberEvent::try_from(e).ok()))
+            .filter_map(|e| e.deserialize().ok())
+            .filter_map(|ev| match ev {
+                RoomMemberEvent::Original(ev) => Some(ev),
+                RoomMemberEvent::Redacted(_) => None,
```

Comment:
Filtering this is incorrect, no? The member event is valid and the member could be joined to the room, it's just the display name and profile info that's removed.

This should at least contain a FIXME explaining the situation, should probably be fixed before 0.5 as well, since this has E2EE implications.

Reply:
Well, @gnunicorn has mentioned now wanting new FIXME comments, it was my first instinct too though ^^

Reply:
I think we can have a FIXME but it should link to an issue.